src/tasks/Microsoft.NET.Sdk.WebAssembly.Pack.Tasks/ConvertDllsToWebCil.cs

```diff
@@ -0,0 +1,90 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.Serialization;
+using System.Runtime.Serialization.Json;
+using System.Text;
+using System.Xml;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using ResourceHashesByNameDictionary = System.Collections.Generic.Dictionary<string, string>;
+
+namespace Microsoft.NET.Sdk.WebAssembly;
+
+public class ConvertDllsToWebCil : Task
+{
+    [Required]
+    public ITaskItem[] Candidates { get; set; }
+
+    [Required]
+    public string OutputPath { get; set; }
+
+    [Required]
+    public bool IsEnabled { get; set; }
+
+    [Output]
+    public ITaskItem[] WebCilCandidates { get; set; }
+
+    protected readonly List<string> _fileWrites = new();
+
+    [Output]
+    public string[]? FileWrites => _fileWrites.ToArray();
+
+    public override bool Execute()
+    {
+        var webCilCandidates = new List<ITaskItem>();
+
+        if (!IsEnabled)
+        {
+            WebCilCandidates = Candidates;
+            return true;
+        }
+
+        for (int i = 0; i < Candidates.Length; i++)
+        {
+            var candidate = Candidates[i];
+
+            var extension = candidate.GetMetadata("Extension");
+            var filePath = candidate.ItemSpec;
+
+            if (!Directory.Exists(OutputPath))
+                Directory.CreateDirectory(OutputPath);
+
+            if (extension == ".dll")
+            {
+                var tmpWebcil = Path.GetTempFileName();
```

Comment:
@lambdageek Since the temp file is always written, what is the reason not to generate directly into output file?

Reply:
The idea is that we don't want to change the modification time of the final output file for incremental builds.  So we write the output to a tmp file and only update the final path if the contents are different.